homework/hm11_bank_card_operations.py

```diff
@@ -0,0 +1,200 @@
+
+TERMINAL_SEPARATOR = '-' * 42
+
+
+class BankCard:
+    bank_name = "StereoBank"
+    currency = "USD"
+
+    def __init__(self, name, surname):
+        self.__balance = 0.00
+        self._is_blocked = False
+        self.name = name
+        self.surname = surname
+        self._card_number = self.generate_card_number()
+        self.__pin = self.generate_pin()
+        self.__cvv = self.generate_cvv()
+        self.__expiration_date = self.calculate_end_date()
+
+    @classmethod
+    def generate_card_number(cls):
+        return random.randint(1000000000000001, 9999999999999999)
+
+    @classmethod
+    def generate_pin(cls):
+        return random.randint(0000, 9999)
+
+    @classmethod
+    def generate_cvv(cls):
+        return random.randint(000, 999)
+
+    @classmethod
+    def calculate_end_date(cls):
+        current_date = datetime.now()
+        end_year = current_date.year + 5
+        end_month = current_date.month
+        return f'{end_month:02d}/{end_year}'
+
+    @classmethod
+    def get_bank_info(cls):
+        return f'Bank: {cls.bank_name}, Currency: {cls.currency}'
+
+    @classmethod
+    def get_currency(cls):
+        return cls.currency
+
+    @classmethod
+    def _check_operation_allowed(cls, balance, amount):
+        if balance - amount < 0:
+            raise (ValueError
+                   ('There are not enough funds to perform this operation.'))
+
+    def open_account(self):
+        self.__balance = 0.00
+
+    def check_balance(self):
+        return self.__balance
+
+    def top_up_balance(self, amount):
+        self.__balance = self.__balance + amount
+        return amount
+
+    def top_down_balance(self, amount):
+        self._check_operation_allowed(self.__balance, amount)
+        self.__balance -= amount
+        return amount
+
+    def transfer_money(self, to_card, amount):
+        self._check_operation_allowed(self.__balance, amount)
+        self.__balance -= amount
+        to_card.top_up_balance(amount)
+        return amount
+
+    def block_card(self):
+        self._is_blocked = True
+
+    def unblock_card(self):
+        self._is_blocked = False
+
+    def is_blocked(self):
+        return self._is_blocked
+
+    def close_account(self, to_card=None):
+        if self.__balance != 0:
+            if to_card:
+                to_card.top_up_balance(self.__balance)
+                self.__balance = 0.00
+                return (f'The account closed successfully. '
+                        f'Remaining balance transferred to '
+                        f'{to_card.name} {to_card.surname}.')
+            else:
+                return ("The account cannot be closed "
+                        "as it has a positive balance. "
+                        "Please provide account to transfer funds.")
+        else:
+            self.__balance = 0.00
+            return "The account closed successfully."
+
+    def get_name(self):
+        return self.name
+
+    def set_name(self, name):
```

Comment:
Here and below: No need in getter or setter if this is public field. Public attributes always could be accessed and modified. You need gettrer and setters for private attriburtes/methods. Also please pay attention that for getters and setters as best practice better to use property decorator - then setter and getter will have the same naming